src/Engine/Physics/Shape.hpp

```diff
@@ -16,70 +16,71 @@ namespace Physics {
+        /// Parameters used to create a sphere shape.
+        struct Sphere {
+            Sphere(float radius)
+                : radius(radius) {}
+            float radius;
+        };
 
-            /// Parameters used to create a plane shape.
-            struct Plane {
-                Plane(const glm::vec3& normal, float planeCoeff)
-                    : normal(normal), planeCoeff(planeCoeff) {}
-                glm::vec3 normal;
-                float planeCoeff;
-            };
+        /// Parameters used to create a plane shape.
+        struct Plane {
+            Plane(const glm::vec3& normal, float planeCoeff)
+                : normal(normal), planeCoeff(planeCoeff) {}
+            glm::vec3 normal;
+            float planeCoeff;
+        };
 
-            /// The various kinds of shapes that are wrapped by %Shape.
-            enum class Kind {
-                Sphere,
-                Plane,
-            };
+        /// The various kinds of shapes that are wrapped by %Shape.
+        enum class Kind {
+            Sphere,
+            Plane,
+        };
 
-            /// Construct a sphere shape.
-            /**
+        /// Construct a sphere shape.
+        /**
              * @param params Sphere specific parameters.
              */
-            ENGINE_API Shape(const Sphere& params);
+        ENGINE_API Shape(const Sphere& params);
 
-            /// Construct a plane shape.
-            /**
+        /// Construct a plane shape.
+        /**
              * @param params Plane specific parameters.
              */
-            ENGINE_API Shape(const Plane& params);
+        ENGINE_API Shape(const Plane& params);
 
-            /// Get the type of wrapped shape.
-            /**
+        /// Get the type of wrapped shape.
+        /**
              * @return The type of shape.
              */
-            ENGINE_API Kind GetKind() const;
+        ENGINE_API Kind GetKind() const;
 
-            /// Get sphere data of the shape.
-            /**
+        /// Get sphere data of the shape.
+        /**
              * @return Sphere data, or nullptr if the shape is not a sphere.
              */
-            ENGINE_API const Sphere* GetSphereData() const;
+        ENGINE_API const Sphere* GetSphereData() const;
 
-            /// Get plane data of the shape.
-            /**
+        /// Get plane data of the shape.
+        /**
              * @return Plane data, or nullptr if the shape is not a plane.
              */
-            ENGINE_API const Plane* GetPlaneData() const;
+        ENGINE_API const Plane* GetPlaneData() const;
 
         private:
-            /// Get the wrapped Bullet shape.
-            /**
+        /// Get the wrapped Bullet shape.
+        /**
              * @return The Bullet shape.
              */
-            btCollisionShape* GetShape() const;
+        btCollisionShape* GetShape() const;
 
-            Shape(const Shape& other) = delete;
+        Shape(const Shape& other) = delete;
 
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Class 'Shape' is unsafe, 'Shape::shape' can leak by wrong usage. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunsafeClassCanLeak)